### PR TITLE
implement declarative platform engine for ISV configuration

### PR DESCRIPTION
**Pull request by hervedombya**
# Veeam Automated Repository Creation

## Overview

This PR adds **automated Veeam repository creation** to the ARTESCA ISV configuration wizard. ARTESCA administrators can now create S3-compatible repositories directly in Veeam Backup & Replication with a single click, eliminating manual configuration steps.

## Problem Statement

**Target Users**: ARTESCA administrators configuring Veeam backup solutions

**Current Pain Point**:

- After configuring ARTESCA (account, buckets, credentials), administrators must manually:
  1. Open Veeam Backup & Replication UI
  2. Create new S3-compatible repository
  3. Configure S3 endpoint, credentials, and bucket
  4. Set up folder structure (`Veeam/Archive/bkp/`, `Veeam/Backup/bkp/Clients/`, etc.)
  5. Configure immutability and storage limits
  6. Test connection and validate setup
- **Time**: ~15 minutes per repository
- **Error-prone**: Manual folder structure often misconfigured

**Solution**:

- One-click repository creation from ARTESCA UI
- Automatic folder structure creation
- Validated configuration with proper immutability and storage limits
- **Time**: ~2 minutes (87% reduction)

### New Feature: Auto-Create Repository

Users now see a new option in the Veeam VBR wizard:

```
☑️ Automatically create Veeam repository
    When enabled, ARTESCA creates the repository in Veeam for you

☑️ Immutable Backup
    ↳ Veeam Immutable retention period: [14] days
```

**What happens when enabled**:

1. **Folder structure creation** (per bucket):

   - `Veeam/Archive/bkp/`
   - `Veeam/Backup/bkp/Clients/`
   - `Veeam/Backup/bkp/Config/`

2. **Repository creation in Veeam**:

   - Calls Veeam REST API: `POST /api/backupInfrastructure/objectStorage/repositories`
   - Polls session until complete (5-minute timeout)
   - Configures: service point, credentials, bucket, storage limits, immutability

3. **Custom summary screen**:
   - Shows repository names with copy buttons
   - Displays immutability status + retention period
   - Provides "next steps" guidance for backup jobs

### Files Changed

**New Components**:

- `components/VeeamRepositoryFields.tsx` - UI for auto-create toggle + immutable period input
- `components/VeeamRepositorySummary.tsx` - Custom summary screen for repository results
- `hooks/useCreateVeeamRepository.ts` - React Query mutation for Veeam API (updated to call real endpoint)
- `utils/ensureHttpsPrefix.ts` - URL normalization helper

**Updated Components**:

- `platforms/veeam-vbr.tsx` - Added folder creation steps + repository mutation
- `hooks/useMutationExecutor.ts` - Added `createVeeamRepository` action
- `engine/types.ts` - Added `createVeeamRepository` to action types

**Tests**:

- `components/__tests__/VeeamRepositoryFields.test.tsx` - 2 tests (toggle + conditional field)
- `components/__tests__/VeeamRepositorySummary.test.tsx` - 2 tests (rendering + interaction)
- `utils/ensureHttpsPrefix.test.ts` - 7 tests (URL normalization edge cases)
- `platforms/__tests__/veeam-vbr.test.tsx` - Minor formatting updates (trailing commas)

**Total**: 10 files changed, 11 new tests added (all passing)

### 1. Repository Configuration UI (`VeeamRepositoryFields`)

- Feature flag check: `useVeeamAutoRepositoryFeature()`
- Conditionally disabled based on account/IAM selection:
  - Enabled when creating new account OR creating new IAM user
  - Disabled when using existing account + existing IAM user (no access keys available)
- Shows immutable retention period input when both auto-create and immutable are enabled
- Validates retention period (1-365 days)

### 2. Folder Creation (Per-Bucket Steps)

Added to `veeam-vbr.tsx` platform definition:

```tsx
perBucketSteps: [
  {
    id: 'veeamArchiveFolder',
    label: 'Create Archive folder: Veeam/Archive/bkp/',
    action: 'putObject',
    when: (form) => form.autoCreateRepository === true,
    variables: (form, bucket, prev) => ({
      s3Client: prev.assumeRole.data,
      Bucket: bucket.name,
      Key: 'Veeam/Archive/bkp/',
      Body: '',
    }),
  },
  // ... similar for Clients/ and Config/ folders
];
```

### 3. Repository Creation Mutation

Added as `additionalMutations` in `veeam-vbr.tsx`:

```tsx
additionalMutations: [
  {
    each: 'buckets',
    steps: [
      {
        id: 'createVeeamRepo',
        label: 'Create Veeam Repository: {{name}}',
        action: 'createVeeamRepository',
        when: (form, bucket, ctx) =>
          form.autoCreateRepository && ctx.needsAccessKey,
        variables: (form, bucket, prev, ctx) => {
          const { kind, count } = calculateStorageConsumptionLimit(
            bucket.capacityBytes,
          );

          return {
            repositoryName: bucket.name,
            servicePoint: ensureHttpsPrefix(ctx._s3ServicePoint),
            accessKey: prev.createAccessKey.data.AccessKey.AccessKeyId,
            secretKey: prev.createAccessKey.data.AccessKey.SecretAccessKey,
            bucketName: bucket.name,
            region: 'us-east-1',
            immutable: form.enableImmutableBackup,
            immutablePeriodDays: form.immutablePeriodDays,
            storageConsumptionLimitKind: kind, // 'TB' or 'PB'
            storageConsumptionLimitCount: count,
          };
        },
      },
    ],
  },
];
```

**Key logic**:

- Runs only when `autoCreateRepository` is enabled AND access keys are available
- Calculates storage limits from bucket capacity (auto-converts bytes → TB or PB)
- Ensures HTTPS prefix on service point URL
- Retrieves access key from previous mutation result

### 4. Veeam API Integration (`useCreateVeeamRepository`)

```tsx
POST /api/veeam-automation/create-s3-repo
Content-Type: application/json
Authorization: Bearer {token}

{
  "repositoryName": "my-repo",
  "servicePoint": "https://s3.artesca.local",
  "accessKey": "AKIAIOSFODNN7EXAMPLE",
  "secretKey": "wJalrXUtnFEMI...",
  "bucketName": "my-repo",
  "region": "us-east-1",
  "immutable": true,
  "immutablePeriodDays": 14,
  "storageConsumptionLimitKind": "TB",
  "storageConsumptionLimitCount": 5
}
```

**Response**:

```json
{
  "repositoryId": "550e8400-e29b-41d4-a716-446655440000",
  "repositoryName": "my-repo",
  "message": "Repository created successfully"
}
```

**Error handling**:

- Catches API errors and displays user-friendly messages
- Timeout after 5 minutes with session polling
- Validates response before proceeding

### 5. Custom Summary Screen (`VeeamRepositorySummary`)

When `autoCreateRepository` is enabled, replaces default summary with:

- **Title**: Success message (singular/plural based on repository count)
- **Repository Names**: Each with copy-to-clipboard button
- **Options Section**:
  - Immutability status (Active/Inactive)
  - Retention period (if immutable)
- **Next Steps**: "Your repository is now available in VBR. Proceed with backup jobs."

### 6. URL Normalization (`ensureHttpsPrefix`)

Ensures service point URLs have proper HTTPS prefix:

```tsx
ensureHttpsPrefix('s3.artesca.local')        → 'https://s3.artesca.local'
ensureHttpsPrefix('https://s3.artesca.local') → 'https://s3.artesca.local'
ensureHttpsPrefix('http://s3.artesca.local')  → 'http://s3.artesca.local'
ensureHttpsPrefix('192.168.1.1:8443')        → 'https://192.168.1.1:8443'
```

## Testing

**11 new tests added, all passing**:

- **`ensureHttpsPrefix.test.ts`** (7 tests): URL normalization edge cases
- **`VeeamRepositoryFields.test.tsx`** (2 tests): UI rendering and conditional display
- **`VeeamRepositorySummary.test.tsx`** (2 tests): Summary rendering and user interaction

**Testing approach**: Behavior-focused, no implementation details

- ✅ Tests verify what users see and interact with
- ✅ Tests validate expected outcomes
- ❌ Tests don't check internal state or implementation

### Feature Flag

- **Flag**: `VeeamAutoRepository`
- **Location**: Runtime configuration (`runtime-app-configuration`)
- **Default**: Must be enabled in deployment config

### Breaking Changes

None - this is purely additive functionality.

### Backend Requirements

This feature requires the veeam-exporter service to be deployed with the repository creation endpoint:

```
POST /api/veeam-automation/create-s3-repo
```

Ensure the endpoint is accessible and configured with:

- Veeam Backup & Replication REST API credentials
- Network access to Veeam server
- Proper CORS configuration for UI calls

### Before (Manual - ~15 minutes)

1. Configure ARTESCA in wizard (5 mins)
2. Copy credentials from summary screen
3. Open Veeam Backup & Replication
4. Navigate to "Add Repository" wizard
5. Select "Object Storage" → "S3 Compatible"
6. Enter service point, credentials, bucket
7. Create folder structure manually
8. Configure immutability settings
9. Set storage quotas
10. Test connection
11. Finish and wait for validation

### After (Automated - ~2 minutes)

1. Configure ARTESCA in wizard with "Auto-create repository" enabled (2 mins)
2. Done! Repository ready in Veeam

## Questions for Reviewers

1. **Error Handling**: Should repository creation failure roll back bucket creation, or leave buckets intact?
2. **Retry Logic**: If repository creation fails, should we provide a retry button in the summary?
3. **Validation**: Should we validate Veeam API connectivity before starting the wizard?
4. **Naming**: Should we allow custom repository names different from bucket names?
5. **Multi-Repository**: For multiple buckets, should repository creation be parallel or sequential?

## Related

- **Feature Flag**: `VeeamAutoRepository` (must be enabled)
- **Backend Service**: veeam-exporter with `/veeam-automation/create-s3-repo` endpoint
- **Documentation**: Update user docs with auto-repository workflow

---

**Breaking Changes**: None  
**Deployment Notes**: Requires veeam-exporter service and feature flag enabled  
**Test Coverage**: 11 new tests, 100% passing

### CODE REVIEW

File: src/react/ISV/engine/types.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ export type RuntimeContext = {
   _platformId: string;
   _bucketTag: string;
   _instanceId: string;
-  _s3ServicePoint: string;
+  _s3ServicePoint?: string;
```
Comment (ZiyangLinScality) [inline src/react/ISV/engine/types.ts:74]: Minor type inconsistency: `_s3ServicePoint` changed from `string` to `string | undefined` in types.ts:74, but `buildRuntimeContext` always provides a default empty string (line 234).
   This makes the `|| ''` fallback in veeam-vbr.tsx redundant.

Comment (hervedombya) [reply src/react/ISV/engine/types.ts:74]: good catch, I I removed the redundant `|| ''` fallback in 65636df52d1fad89351194df675aa3614b0e9172

**Review by ZiyangLinScality — APPROVED** [review]
LGTM
